### ISSUE TITLE
change PR to "waiting on author" after "changes requested" review  #1373

### ISSUE THREAD

**Issue by tgross35** [rust-lang/triagebot#1857]
This was requested in https://github.com/rust-lang/triagebot/issues/1373 and added in https://github.com/rust-lang/triagebot/pull/1379 (cc @mibac138), but it seems like this must have broken at some point.

**Comment by ehuss** [comment]
Can you say why it does not appear to be working? I tested it locally and it seems to be working for me. Is there a PR that shows an example?

**Comment by tgross35** [comment]
I just tried it with a handful of PRs in libc and wound up needing to add the label myself, e.g. https://github.com/rust-lang/libc/pull/3962 or https://github.com/rust-lang/libc/pull/3261. An example in rust-lang/rust https://github.com/rust-lang/rust/pull/132939.

**Comment by tgross35** [comment]
Although it looks like maybe it works at least some of the time? https://github.com/rust-lang/rust/pull/132791

**Comment by ehuss** [comment]
The feature is not enabled in the libc repository: https://github.com/rust-lang/libc/blob/main/triagebot.toml

See https://forge.rust-lang.org/triagebot/review-submitted.html

**Comment by tgross35** [comment]
Okay so it's still not working, but something must be wrong with GitHub here. I approved https://github.com/rust-lang/libc/pull/4028 so it should say "approved" on the PR list but it does not:

![Image](https://github.com/user-attachments/assets/88e77047-e18f-4fe9-b579-c9640a755a24)

Additionally, going reviews->approved review shows nothing recent (even though I have been using the approval UI for a while)

![Image](https://github.com/user-attachments/assets/29bd379d-bd74-4371-8fbd-c32c5c7574ee)

Seems this might be a bigger issue https://stackoverflow.com/questions/75637877/github-filtering-not-showing-approved-reviews

**Comment by ehuss** [comment]
I'm not sure why GitHub is not showing the pulls (maybe because reviews aren't required?). However, I don't think that is related here. I believe the labels weren't changed on https://github.com/rust-lang/libc/pull/4028 because you were not the assignee (that looks to be changed a minute after approving).

**Comment by tgross35** [comment]
I did try reassigning to myself then reapproving, still no luck - I'll have to wait for a fresh one to roll in and try assigning before I do anything. But I don't think GH requires the assignee be the same as the approver, e.g. here is one in rust-lang/rust where I approved and then assigned to myself https://github.com/rust-lang/rust/pull/132914 and it does show "approved" in the PR list. `gh pr view 132914 --repo https://github.com/rust-lang/rust --json reviewDecision` also reports `"reviewDecision": "APPROVED"` for this one, so something seems to be related between `reviewDecision` and what shows in the PR list (by extrapolation I am assuming this relates to what triggers triagebot, though I haven't traced back via the REST API).
